File: remotecommands/tealium-react-braze/ios/TealiumReactBraze.m
// TealiumReactBraze.m

#import "React/RCTBridgeModule.h"


@interface RCT_EXTERN_MODULE(TealiumReactBraze, NSObject)

    RCT_EXTERN_METHOD(initialize)

    RCT_EXTERN_METHOD(configure:(NSDictionary *)options)

@end

File: remotecommands/tealium-react-braze/ios/TealiumReactBraze.swift
//
//  TealiumReactBraze.swift
//  tealium-react-braze
//
//  Created by Tyler Rister on 03/25/2022.
//

import Foundation
import React
import tealium_react_native

@objc(TealiumReactBraze)
class TealiumReactBraze: NSObject, RCTBridgeModule {
    static func moduleName() -> String! {
        "TealiumReactBraze"
    }
    let factory = BrazeRemoteCommandWrapper()
    
    @objc
    static func requiresMainQueueSetup() -> Bool {
        return false
    }
    
    override init() {
        super.init()
        TealiumReactNative.registerRemoteCommandFactory(factory)
    }
    
    @objc(initialize)
    public func initialize() {
        TealiumReactNative.registerRemoteCommandFactory(factory)
    }
    
}
